MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -601,4 +601,16 @@ open class RouteStepProgress: NSObject {
         guard visualInstructionIndex < instructionsDisplayedAlongStep.count else { return nil }
         return instructionsDisplayedAlongStep[visualInstructionIndex]
     }
+    
+    @objc override open class func keyPathsForValuesAffectingValue(forKey key: String) -> Set<String> {
```

Comment:
This is fine, but it’s a bit cleaner to implement separate [`keyPathsAffectingValueForRemainingVisualInstructions` and `keyPathsAffectingValueForRemainingSpokenInstructions`](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/KeyValueObserving/Articles/KVODependentKeys.html#//apple_ref/doc/uid/20002179-SW2) properties.